cukinia

```diff
@@ -363,16 +363,45 @@ _cukinia_process()
 {
 	local pname="$1"
 	local puser="$2"
-	local ret
+	local proc_name
+	local proc_uid
 
 	_cukinia_prepare "Checking process \"$pname\" as ${puser:-any user}"
-	ret=$(ps -a -o user,comm |
-			awk '($2 == "'$pname'") && ($1 ~ /^'${puser:-.*}'$/) {
-			    print;
-			}')
-	[ -n "$ret" ] && return 0 || return 1
-}
 
+	if [ ! -z $puser ]; then
+		puser_id=`id -u $puser &>/dev/null`
+		# check if owner exist, return 1 then
+		if [ $? -ne 0 ]; then
+			return 1
+		fi
+	fi
+
+	# loop in /proc to get process name and optionally process owner
+	for p in `ls /proc/ | grep -Eo ''[[:digit:]]*''`; do
+		# ensure that a status file does exist
```

Comment:
The **for** loop and **if** statement can be shortened in *one single* line.

``` bash
for proc in /proc/[0-9]*/status; do
    # ...
done
```

Reply:
yep thanks.